src/silo/common/table_reader.cpp

```diff
@@ -26,24 +34,38 @@ silo::TableReader::TableReader(
       where_clause(where_clause),
       order_by_clause(order_by_clause) {}
 
-std::optional<std::string> silo::TableReader::nextKey() {
-   if (!current_chunk) {
-      return std::nullopt;
-   }
-
-   return current_chunk->GetValue(0, current_row_in_chunk).GetValue<std::string>();
-}
-
-void silo::TableReader::read() {
+size_t silo::TableReader::read() {
    loadTable();
    assert(query_result->ColumnCount() == column_functions.size() + 1);
-   while (nextKey()) {
-      for (size_t column_idx = 0; column_idx < column_functions.size(); column_idx++) {
-         column_functions.at(column_idx)
-            .function(current_row, current_chunk->GetValue(column_idx + 1, current_row_in_chunk));
+   while (true) {
+      current_chunk = query_result->Fetch();
+      if (!current_chunk) {
+         break;
       }
-      advanceRow();
-   }
+      if (current_chunk->size() == 0) {
+         continue;
+      }
+      current_chunk_size = current_chunk->size();
+
+      tbb::parallel_for(
+         tbb::blocked_range<uint32_t>(0, column_functions.size()),
+         [&](const auto& local) {
+            for (size_t column_idx = local.begin(); column_idx != local.end(); column_idx++) {
+               auto column = current_chunk->data[column_idx + 1];
```

Comment:
Is this copy really necessary? Does anything speak against turning this into a reference (possible const?, if GetValue allows it)